packages/datadog-instrumentations/src/express.js

```diff
@@ -129,7 +145,37 @@ addHook({ name: 'express', versions: ['>=4.0.0 <4.3.0'] }, express => {
   return express
 })
 
-addHook({ name: 'express', versions: ['>=4.3.0'] }, express => {
+addHook({ name: 'express', versions: ['>=4.3.0 <5.0.0'] }, express => {
   shimmer.wrap(express.Router, 'process_params', wrapProcessParamsMethod(2))
   return express
 })
+
+addHook({ name: 'express', file: ['lib/express.js'], versions: ['>=5.0.0'] }, express => {
+  shimmer.wrapFunction(express, function (original) {
+    const app = original.call(this, arguments)
+    const requestProto = Object.getPrototypeOf(app.request)
+    const requestDescriptor = Object.getOwnPropertyDescriptor(requestProto, 'query')
+
+    shimmer.wrap(requestDescriptor, 'get', function (originalGet) {
+      return function wrappedGet () {
+        const query = originalGet.apply(this, arguments)
+
+        if (queryParserReadCh.hasSubscribers) {
+          const abortController = new AbortController()
+
+          queryParserReadCh.publish({ req: this, res: this.res, query, abortController })
+
+          if (abortController.signal.aborted) return
```

Comment:
This is the right place to make iast work, because we need to taint the values each time the request is called, but is this the right place to use the abortController and call to the waf? Calling to the waf each time that `req.query` code is in the app, could be quite expensive.

Reply:
yeah maybe the best thing to do would be to just remove the waf call here, and instead call it in the router handle request hook

Reply:
I agree with you about how expensive this operation could be, but just to clarify:

1. In express v4, we don’t taint the query string using express’ [query method](https://github.com/DataDog/dd-trace-js/blob/master/packages/datadog-instrumentations/src/express.js#L92) but instead rely on the `qs` [parse function.](https://github.com/DataDog/dd-trace-js/blob/9d0f9fa7e53c56a3e01d5524e7dd7b61354ca1cd/packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js#L49)
2. IMO, we should harmonize how we taint query strings between express v4 and express v5. This could be achieved either by using both `qs` and `querystring` (though this might lead to false positives since we can’t confirm if the source is from express) or by keeping the getter instrumentation here and using it for IAST, as you suggested. 
We could also apply the same approach to express v4 and then (remove ?) the `qs` and `querystring` instrumentation altogether.

Reply:
Definitely, we can remove the `qs` instrumentation to do it in the same way in express v4 and v5. 
I can't recall why are we instrumenting the qs and not reading the query in other position of the code, I think that we can move all to express instrumentation.

Not necessarily in this PR, if you see that could be done easily, go ahead, but we can change express v4 stuff in other PR.

Reply:
FYI, it is possible to configure the express5 query parser:

> The simple query parser is based on Node’s native query parser, [querystring](http://nodejs.org/api/querystring.html).
> The extended query parser is based on [qs](https://www.npmjs.org/package/qs).

`simple` is configured by default, but `extended` could be configured and it will use `qs` lib.

But it doesn't matter if we move the instrumentation to express...